server/mvc/models/User/RegionProfileMapper.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+var Model;
+
+var getModel = function(thinky){
+  if (Model === undefined) {
+    Model = thinky.createModel("RegionProfiles", {
+      userProfileId: thinky.type.number().integer().default(null).min(1).required().allowNull(false),
+      userId: thinky.type.number().integer().default(null).min(1).required().allowNull(false),
+      region: thinky.type.string().default(null).min(1).required().allowNull(true).validator(function(homeRegion){
+        return (
+          homeRegion === null
+          || (
+            typeof(homeRegion) == "string"
+            && (["us", "eu"].indexOf(homeRegion) > -1)
+          )
+        );
+      }),
+      characterName: thinky.type.string().default(null).required().allowNull(true).validator(function(nickname){
```

Comment:
We should probably update the logic in this validator. Currently, users can use e.g. spaces and weird symbols in their names.

The logic should be something like:

- Must be a string
- Must be a least 2 characters long
- Must contain uppercase and lowercase UTF-8 letters
- Must not contain non-UTF-8-letters, i.e. symbols and spaces.

**Suggested [XRegExp](http://xregexp.com/plugins/) pattern**:

```javascript
XRegExp(^\\p{L}{2;}$); // Match a unicode word
```

However, as you suggest we use the battletag as the nickname ([here](https://github.com/Starbow/website/pull/19/files#diff-16e99882f7e93de1cf5fd6a0ba45a89fR99)), which is comprised of a name, a hashtag, and a number (e.g. foo#123), we'll need to allow this specific case. E.g.:

```javascript
XRegExp(^\\p{L}{2;}(\#(\d+))?$); // Match a unicode word
```